core/src/main/java/org/frankframework/validation/XSD.java

```diff
@@ -246,27 +252,52 @@ public int compareTo(IXSD other) { // CompareTo is required for WSDL generation
 		return compareToByReferenceOrContents(other);
 	}
 
-	protected int compareToByReferenceOrContents(IXSD x) {
-		return compareToByContents(x);
+	protected int compareToByReferenceOrContents(IXSD other) {
+		return compareToByContents(other);
 	}
 
 	@Override
-	public int compareToByContents(IXSD x) {
+	public int compareToByContents(IXSD other) {
+		Integer cachedResult = compareByContentsCache.get(System.identityHashCode(other));
+		if (cachedResult != null) {
+			return cachedResult;
+		}
 		try {
 			InputSource control = new InputSource(getReader());
-			InputSource test = new InputSource(x.getReader());
+			InputSource test = new InputSource(other.getReader());
 			Diff diff = new Diff(control, test);
 			if (diff.similar()) {
-				return 0;
+				return cacheCompareContentsResult(other, 0);
 			}
 			// When "diff" says they're different we still need to order them for the "compareTo" result, so we make strings and compare the strings.
-			return StreamUtil.readerToString(getReader(), "\n", false).compareTo(StreamUtil.readerToString(x.getReader(), "\n", false));
+			int compareResult = StreamUtil.readerToString(getReader(), "\n", false).compareTo(StreamUtil.readerToString(other.getReader(), "\n", false));
+			return cacheCompareContentsResult(other, compareResult);
 		} catch (Exception e) {
 			LOG.warn("Exception during XSD compare", e);
 			return 1;
 		}
 	}
 
+	/**
+	 * The operation {@link #compareToByContents(IXSD)} is quite expensive, and sometimes we compare the very same XSD documents
+	 * multiple times. Caching the results gives a minor speedup on startup of the framework.
+	 * <br/>
+	 * The cache is keyed by the {@link System#identityHashCode} of the other XSD against which the comparison was made and stores the previous content-comparison result.
+	 * The corresponding cache of the other XSD is also updated, keyed by our own identityHashCode. The identityHashCode is chosen because it's fast, and because
+	 * it is guaranteed to be different for instances which might otherwise calculate the same {@link #hashCode()}.
```

Comment:
Misschien ook nog even toevoegen dat je dit ook aan de andere kant van de compare bijhoudt (blokje in de if instanceof)

Reply:
Gedaan.